vms/evm/warp/backend.go

```diff
@@ -0,0 +1,211 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package warp
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/libevm/log"
+
+	"github.com/ava-labs/avalanchego/cache"
+	"github.com/ava-labs/avalanchego/cache/lru"
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/network/p2p/acp118"
+	"github.com/ava-labs/avalanchego/snow/consensus/snowman"
+	"github.com/ava-labs/avalanchego/vms/evm/uptimetracker"
+	"github.com/ava-labs/avalanchego/vms/platformvm/warp/payload"
+
+	avalancheWarp "github.com/ava-labs/avalanchego/vms/platformvm/warp"
+)
+
+var (
+	_                         Backend = (*backend)(nil)
+	ErrValidateBlock                  = errors.New("failed to validate block message")
+	ErrVerifyWarpMessage              = errors.New("failed to verify warp message")
+	errParsingOffChainMessage         = errors.New("failed to parse off-chain message")
+
+	messageCacheSize = 500
+)
+
+type BlockClient interface {
+	GetAcceptedBlock(ctx context.Context, blockID ids.ID) (snowman.Block, error)
+}
+
+// Backend tracks signature-eligible warp messages and provides an interface to fetch them.
+// The backend is also used to query for warp message signatures by the signature request handler.
+type Backend interface {
+	// AddMessage signs [unsignedMessage] and adds it to the warp backend database
+	AddMessage(unsignedMessage *avalancheWarp.UnsignedMessage) error
```

Comment:
I was trying to figure out what this abstraction was - we sign messages that are defined as valid - but we also have a k/v store of stuff we're always willing to sign (so a db and a verifier/signer?). Upon reading its usage - I don't think we actually use this in prod code though - so this is just an API for testing. Can we see if we can simplify and remove this?